transformer_engine/jax/csrc/utils.cpp

```diff
@@ -0,0 +1,37 @@
+/*************************************************************************
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See LICENSE for license information.
+ ************************************************************************/
+#include <cassert>
+#include <cuda_runtime_api.h>
+
+#include "utils.h"
+
+namespace transformer_engine {
+namespace jax {
+
+int GetCudaRuntimeVersion() {
+    int ver = 0;
+    NVTE_CHECK_CUDA(cudaRuntimeGetVersion(&ver));
+    return ver;
+}
+
+int GetDeviceComputeCapability(int gpu_id) {
+    int max_num_gpu = 0;
+    NVTE_CHECK_CUDA(cudaGetDeviceCount(&max_num_gpu));
+    assert(gpu_id < max_num_gpu);
+
+    int major = 0;
+    NVTE_CHECK_CUDA(cudaDeviceGetAttribute(&major, cudaDevAttrComputeCapabilityMajor, gpu_id));
```

Comment:
Since we don't cache these major/minor attributes in the FW, have we confirmed that this call is free(ish)? Previously we observed a non-negligible perf regression upon calling `cudaGetDeviceProperties`. Even though this call queries just a single value, it would be good to double check before merging. @jeng1220

Reply:
It is fine. `cudaDeviceGetAttribute` only takes 0.1 microseconds. The reason of `cudaGetDeviceProperties` taking time is that it needs to return a lot of attributes. `cudaDeviceGetAttribute` only returns 1 attribute.